corehq/motech/utils.py

```diff
@@ -202,3 +202,26 @@ def copy_api_auth_settings(connection):
     connection.pass_credentials_in_header = api_settings.pass_credentials_in_header
 
     connection.save()
+
+
+def api_setting_matches_preset(connection):
+    def split_url(url):
+        if not url:
+            return None
+        try:
+            return url.split(connection.url.rstrip('/'))[1]
+        except IndexError:
+            return None
```

Comment:
~Are we on Python 3.9 yet?~ Only Staging is on Python 3.9. Pity. [PEP 616](https://www.python.org/dev/peps/pep-0616/) was made for this.

```python
    def get_endpoint(url):
        base_url = connection.url.rstrip('/')
        return url and url.removeprefix(base_url)
```